src/applications/vaos/appointment-list/redux/actions.js

```diff
@@ -607,8 +602,12 @@ export function fetchConfirmedAppointmentDetails(id, type) {
       }
 
       facilityId = getVAAppointmentLocationId(appointment);
-      facility = state.appointments.facilityData?.[facilityId];
+      facility =
+        state.appointments.facilityData?.[facilityId] ||
+        appointment.vaos.facilityData;
```

Comment:
Similarly, grab the facility data from the appointment, so we don't unnecessarily fetch it